verification.py

```diff
@@ -758,6 +759,271 @@ def test_sweep_ex1(self):
+                num_sites = int(tokens[ind + 2]) + 1
+            elif tokens[ind] == "-SF":
+                num_sweeps += 1
+                end_time_lst.append(float(tokens[ind + 1]))
+                end_frequency_lst.append(float(tokens[ind + 2]))
+            elif tokens[ind] == "-Sp":
+                sel_pos = float(tokens[ind + 1])
+            elif tokens[ind] == "-SAA":
+                sAA = float(tokens[ind + 1])
+            elif tokens[ind] == "-SaA":
+                saA = float(tokens[ind + 1])
+            elif tokens[ind] == "-N":
+                refsize = int(tokens[ind + 1])
+            elif tokens[ind] == "-seed":
+                rand_seed = float(tokens[ind + 1])
+            else:
+                pass
+
+        # check if h = 0.5
+        if abs(saA * 2 - sAA) > 1e-5:
+            logging.warning(
+                "If 2 * saA is not equal to sAA, saA is set to sAA / 2,"
+                "that is h can only be 0.5 in msprime"
+            )
+            saA = sAA / 2.0
+
+        return {
+            "nsam": nsam,
+            "nrep": nrep,
+            "num_sweeps": num_sweeps,
+            "end_time_lst": end_time_lst,
+            "end_frequency_lst": end_frequency_lst,
+            "Ne": refsize,
+            "alpha": sAA,
+            "theta": theta,
+            "rho": rho,
+            "num_sites": num_sites,
+            "sel_pos": sel_pos,
+            "rand_seed": rand_seed,
+        }
+
+    def _msms_parms_to_run_msp(self, params):
+        """ run smulation for a single sample and return a tree sequence.
+            this method is called by `_run_msp_sample_stats` in a loop to
+            generate nrep samples.
+        """
+        Ne_temp = None
+        if params["num_sweeps"] > 0:
+            model = [None]
+            for i in range(params["num_sweeps"]):
+                temp_model = msprime.SweepGenicSelection(
+                    position=params["sel_pos"],
+                    end_frequency=params["end_frequency_lst"][i],
+                    start_frequency=1.0 / 2 / params["Ne"],  # should be small
+                    alpha=params["alpha"],
+                    dt=1.0 / 40 / params["Ne"],
+                )  # should be small enough?
+                t_start = params["end_time_lst"][i]
+                model.append((t_start, temp_model))
+            model.append((None, None))
+
+            # To circumvent huson/genicselection scaling inconsistency
+            # the scaling is mimicing verficatin.py - DiscoalTest class -
+            # _discoal_str_to_simulation methods.
+            recombination_rate = params["rho"] / (params["num_sites"] - 1)
+            mutation_rate = params["theta"] / params["num_sites"]
+            Ne_temp = 0.25
+
+        else:
+            model = "hudson"
+            # To circumvent huson/genicselection scaling inconsistency
+            recombination_rate = params["rho"] / params["num_sites"] / 4 / params["Ne"]
```

Comment:
It'd be good to factor the scaling factor out here rather than repeating the same calculation twice. Also, I find it difficult to read things like ``1 / n / x`` isn't that said more plainly as x / n?